store/src/store.rs

```diff
@@ -142,9 +142,8 @@ pub trait ChainStore<'a>: Send + Sync + Sized {
             )
             .take_while(|(key, _)| key.starts_with(prefix))
             .map(|(_key, value)| {
-                let reader =
-                    packed::TransactionViewReader::from_slice_should_be_ok(&value.as_ref());
-                reader.hash().to_entity()
+                // transaction hash is serialized as TransactionView's first field, we use partial reads here to avoid the extra cost of deserialization.
+                packed::Byte32::new_unchecked(value[16..48].to_owned().into())
```

Comment:
If the purpose of this PR is to skip all checks, the above code is equal to `packed::TransactionViewReader::new_unchecked(&value.as_ref()).hash().to_entity()`.

By the way, I think at least the length should be checked to avoid `panic`.
The deserialization only checks all the structure of data -- all length are right.
`Reader` doesn't do `clone`.

Reply:
resolved, please review again, and I created a new PR: https://github.com/nervosnetwork/ckb/pull/2579, if it's approved, we can close this PR.